### ISSUE TITLE
a wrong about tf.losses.softmax_cross_entropy

### ISSUE THREAD

**Issue by sanlb** [Goldesel23/Temporal-Ensembling-for-Semi-Supervised-Learning#6]
tf.losses.softmax_cross_entropy is softmax(output) and then cross_entropy, while the model's last layer--dense also has the softmax activation. It means that the final loss will softmax twice. is it wrong?

**Comment by tensorfreitas** [comment]
I don't think so. The `tf.losses.softmax_cross_entropy` only calculates the loss bewtween the one-hot labels and the logits... It doesn't apply the activation a second time.

I've confirmed on the code and it does not apply the softmax.
